frontend/src/ScoreTable.js

```diff
@@ -1,87 +1,60 @@
-import React, { Component } from 'react'
+import React, { useEffect, useState } from 'react'
 import ReactTable from 'react-table'
 
-/**
- * A wrapper for the react table with common table functionality and styling.
- */
-class ScoreTable extends Component {
-  constructor(props) {
-    super(props)
+const ScoreTable = props => {
+  const { noun, multi, data = [], onSelect } = props
+  const [selected, setSelected] = useState(multi ? {} : null)
+  const [sorted, setSorted] = useState(props.sorted)
 
-    this.state = {
-      selected: this.props.multi ? {} : null,
-      sorted: this.props.sorted,
-    }
+  useEffect(() => {
+    const selected = multi ? {} : null
+    setSelected(selected)
+    onSelect && onSelect(selected)
+  }, [data])
 
-    this.resort = this.resort.bind(this)
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.data !== this.props.data) {
-      // TODO: Switch to functional component and use useEffect(() => {...}, [])
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({ selected: this.props.multi ? {} : null })
-      if (this.props.onSelect) {
-        this.props.onSelect(this.props.multi ? {} : null)
-      }
-    }
-  }
-
-  resort() {
-    this.setState(state => ({
-      sorted: state.sorted.slice(),
-    }))
-  }
-
-  render() {
-    return (
-      <div>
-        <ReactTable
-          className="mb-2"
-          {...this.props}
-          showPagination={false}
-          resizable={false}
-          style={{ maxHeight: 400 }}
-          getTrProps={(_, rowInfo) => {
-            const { selected } = this.state
-            const { multi, onSelect } = this.props
-            const { index, original, row } = rowInfo
-            const noRow = multi ? index in selected : index === selected
-            if (rowInfo && row) {
-              return {
-                onClick: () => {
-                  if (multi) {
-                    if (noRow) {
-                      delete selected[index]
-                    } else {
-                      selected[index] = rowInfo
-                    }
-                    onSelect && onSelect(selected)
-                    return this.setState({ selected: { ...selected } })
-                  }
-                  onSelect && onSelect(noRow ? null : original.key)
-                  return this.setState({ selected: noRow ? null : index })
-                },
-                className: noRow ? 'selected' : '',
+  const getTrProps = (_, rowInfo) => {
+    const { index, original, row } = rowInfo
+    const noRow = multi ? index in selected : index === selected
+    return rowInfo && row
+      ? {
+          onClick: () => {
+            if (multi) {
+              if (noRow) {
+                delete selected[index]
+              } else {
+                selected[index] = rowInfo
               }
+              onSelect && onSelect(selected)
```

Comment:
I prefer if statements if we are calling functions / things with side effects. The and operators make it seem like this is just a boolean comparison when in reality state is changing